ScratchWikiSkinTemplate.php

```diff
@@ -103,6 +103,37 @@ public function execute() {
 					</div>
 				</form>
 			</li>
+			<?php
+			//  Start dropdown here so we know its output before it saves to the page, and we can use it earlier.
+			$swsDropdownItems = "";
+			foreach ($this->data['personal_urls'] as $key => $tab) { 
+					if (!array_key_exists('href', $tab)) {
+						if (!array_key_exists('class', $tab)) {
+							$tab['class'] = '';
+						}
+						
+						$tab['class'] .= 'no-link';
+					}
+					if(isset($tab["class"]) && gettype($tab["class"]) == "array" && $tab["class"][0] == "mw-echo-alert"){
+						$swsUnread = true;
```

Comment:
a better name for this variable would be good; also you should define it as false before the loop, otherwise you risk getting a "variable undefined" error further down

Reply:
There is no risk of a "variable undefined" error. The place it's read from uses `isset()` and it's deliberately *not* defined as false because that would break the isset check.

Reply:
Fair enough